### PR TITLE
Port some missing metrics from 3.7 for ConnectionPool to devel

### PR DESCRIPTION
**Pull request by neunhoef**
This adds some metrics which do already exist in 3.7 in the
ConnectionPool to devel. Tests are adjusted using mock servers.

### Scope & Purpose

*(Please describe the changes in this PR for reviewers - **mandatory**)*

- [*] :hankey: Bugfix (requires CHANGELOG entry)
- [*] :hankey: Trivial forward port of some metrics.

#### Backports:

- [*] No backports required

### Testing & Verification

*(Please pick either of the following options)*

- [*] This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: arangod/Network/ConnectionPool.cpp [hunk 1]
```diff
@@ -39,7 +39,23 @@ using namespace arangodb::fuerte::v1;
...
+          std::string("arangodb_connection_pool_nr_conns_") + _config.name, uint64_t(0), "Current number of connections in pool")),
+      _successSelect(
+        _config.metricsFeature.counter(
+        std::string("arangodb_connection_pool_success_select_") + _config.name, 0, "Total number of successful connection leases")),
+      _noSuccessSelect(
+        _config.metricsFeature.counter(
+          std::string("arangodb_connection_pool_no_success_select_") + _config.name, 0, "Total number of failed connection leases")),
+      _connectionsCreated(
+        _config.metricsFeature.counter(
+          std::string("arangodb_connection_pool_conns_created_") + _config.name, 0, "Number of connections created")),
```
Comment (jsteemann) [inline arangod/Network/ConnectionPool.cpp:54]:
```suggestion
          std::string("arangodb_connection_pool_connections_created_") + _config.name, 0, "Total number of connections created")),
```

File: arangod/Network/ConnectionPool.cpp [hunk 2]
```diff
@@ -39,7 +39,23 @@ using namespace arangodb::fuerte::v1;
...
+      _loop(config.numIOThreads, config.name),
+      _totalConnectionsInPool(
+        _config.metricsFeature.gauge(
+          std::string("arangodb_connection_pool_nr_conns_") + _config.name, uint64_t(0), "Current number of connections in pool")),
+      _successSelect(
+        _config.metricsFeature.counter(
+        std::string("arangodb_connection_pool_success_select_") + _config.name, 0, "Total number of successful connection leases")),
+      _noSuccessSelect(
+        _config.metricsFeature.counter(
+          std::string("arangodb_connection_pool_no_success_select_") + _config.name, 0, "Total number of failed connection leases")),
```
Comment (jsteemann) [inline arangod/Network/ConnectionPool.cpp:51]:
```suggestion
          std::string("arangodb_connection_pool_leases_failed_") + _config.name, 0, "Total number of failed connection leases")),
```

File: arangod/Network/ConnectionPool.cpp [hunk 3]
```diff
@@ -39,7 +39,23 @@ using namespace arangodb::fuerte::v1;
...
 ConnectionPool::ConnectionPool(ConnectionPool::Config const& config)
     : _config(config), 
-      _loop(config.numIOThreads, config.name) {
+      _loop(config.numIOThreads, config.name),
+      _totalConnectionsInPool(
+        _config.metricsFeature.gauge(
+          std::string("arangodb_connection_pool_nr_conns_") + _config.name, uint64_t(0), "Current number of connections in pool")),
+      _successSelect(
+        _config.metricsFeature.counter(
+        std::string("arangodb_connection_pool_success_select_") + _config.name, 0, "Total number of successful connection leases")),
```
Comment (jsteemann) [inline arangod/Network/ConnectionPool.cpp:48]:
```suggestion
        std::string("arangodb_connection_leases_successful_") + _config.name, 0, "Total number of successful connection leases")),
```

File: arangod/Network/ConnectionPool.cpp [hunk 4]
```diff
@@ -39,7 +39,23 @@ using namespace arangodb::fuerte::v1;
 
 ConnectionPool::ConnectionPool(ConnectionPool::Config const& config)
     : _config(config), 
-      _loop(config.numIOThreads, config.name) {
+      _loop(config.numIOThreads, config.name),
+      _totalConnectionsInPool(
+        _config.metricsFeature.gauge(
+          std::string("arangodb_connection_pool_nr_conns_") + _config.name, uint64_t(0), "Current number of connections in pool")),
```
Comment (jsteemann) [inline arangod/Network/ConnectionPool.cpp:45]:
```suggestion
          std::string("arangodb_connection_connections_current_") + _config.name, uint64_t(0), "Current number of connections in pool")),
```

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13421/

**Pull request merged**